packages/runtime/lifecycle/src/apis/runtimeLifecycle.ts

```diff
@@ -0,0 +1,36 @@
+
+
+import * as TE from 'fp-ts/lib/TaskEither.js'
+import * as F from 'fp-ts/lib'
+import { WalletAPI } from '@marlowe.io/wallet/api';
+import * as O from 'fp-ts/lib/Option.js'
+import { CreateRequest, ProvideInput, TxAPI } from './tx.js';
+import { AssetId, ContractId, PayoutAvailable, PayoutId, PayoutWithdrawn } from '@marlowe.io/runtime-core';
+import { DecodingError } from '@marlowe.io/adapter/codec';
+
+export type RuntimeLifecycle =
+  { wallet : WalletAPI 
+    contracts : {
+      create      : (payload: CreateRequest)   => TE.TaskEither<Error | DecodingError,ContractId>
+      applyInputs : (contractId: ContractId)  => (provideInput : ProvideInput) => TE.TaskEither<Error | DecodingError,ContractId>
+    }
+    payouts :  
+      { available : (filtersOption : O.Option<Filters>) => TE.TaskEither<Error | DecodingError,PayoutAvailable[]>
```

Comment:
If this is the end user API, cant we use a `available: (filters?: Filters) => Promise<PayoutAvailable[]>`?

With typedoc we can use the [throws](https://typedoc.org/tags/throws/) to signal the `DecodingError` case

Reply:
cool that was one my questions for a un-fp-ts interface